django_project/minisass_frontend/src/components/ObservationDetails/index.tsx

```diff
@@ -84,7 +84,7 @@ const ObservationDetails: React.FC<ObservationDetailsProps> = ({
   }
 
   const fetchObservations = async () => {
-    const url = `${OBSERVATION_LIST_URL}/?site_id=${observationDetails.site.gid}&recent=False`
+    const url = `${OBSERVATION_LIST_URL}/?site_id=${observationDetails.site?.gid}&recent=False`
```

Comment:
@zamuzakki I think it's better to check if siteId exists before making the request to avoid unnecessary network calls